sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -693,6 +719,12 @@ public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
 
     @Override
     public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offersToProcess) {
+        if (!apiServerDisabled.get() && (apiServer == null || !apiServer.isStarted())) {
+            LOGGER.info("Waiting for API Server to start ...");
+            declineOffers(driver, Collections.emptyList(), offersToProcess);
+            return;
```

Comment:
I'm not really sure what's causing that delay for the API server. Some of the tasks actually come up and are COMPLETE before that API server starts responding. Anyway, didn't mean to hijack this PR with it, but I thought it might be relevant.